internal/manifest/oci_index.go

```diff
@@ -84,6 +84,54 @@ func (index *OCI1IndexPublic) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (index *OCI1Index) EditInstances(updates []ListEdit) error {
+	if len(updates) != len(index.Manifests) {
+		//return fmt.Errorf("incorrect number of update entries passed to OCI1Index.UpdateInstances: expected %d, got %d", len(index.Manifests), len(updates))
+		if len(updates) > len(index.Manifests) {
+			newIndexes := make([]imgspecv1.Descriptor, len(updates))
+			copy(newIndexes, index.Manifests)
+			index.Manifests = newIndexes
+		}
+	}
+	for i, editInstance := range updates {
+		switch editInstance.ListOperation {
+		case ListOpUpdate:
+			if err := updates[i].Digest.Validate(); err != nil {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances contained an invalid digest: %w", i+1, len(updates), err)
+			}
+			index.Manifests[i].Digest = updates[i].Digest
+			if updates[i].Size < 0 {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had an invalid size (%d)", i+1, len(updates), updates[i].Size)
+			}
+			index.Manifests[i].Size = updates[i].Size
+			if updates[i].MediaType == "" {
+				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had no media type (was %q)", i+1, len(updates), index.Manifests[i].MediaType)
+			}
+			index.Manifests[i].MediaType = updates[i].MediaType
+			if updates[i].Platform != nil {
+				index.Manifests[i].Platform = updates[i].Platform
+			}
+			if updates[i].Annotations != nil {
+				index.Manifests[i].Annotations = *updates[i].Annotations
+			}
+		case ListOpIgnore:
+			continue
+		case ListOpRemove:
+			index.Manifests = removeManifests(index.Manifests, i)
+		default:
+			return fmt.Errorf("internal error: invalid operation")
+		}
+	}
+	return nil
+}
+
+// removeManifests is internal function which takes the original slice and an index,
+// as a result it will return slice of manifests with element removed and order kept
+// intact.
+func removeManifests(slice []imgspecv1.Descriptor, s int) []imgspecv1.Descriptor {
+	return append(slice[:s], slice[s+1:]...)
```

Comment:
BTW I’d prefer this to use https://pkg.go.dev/golang.org/x/exp/slices#Delete instead of open-coding.